File: Auxiliary/gclib/examples/cpp/examples/contour_example.cpp
/** @addtogroup cpp_examples
  * @{
  */
  
/*! \file contour_example.cpp

See contour() for implementation of logic

*/
#include "examples.h"

#include <iostream> //std::cout
using namespace std;

//! Main function for Contour Example
/*!
\link contour_example.cpp \endlink takes three arguments at the command 
line: an IP Address to a Galil controller and a two text files to hold the 
positional data for two axes.
*/
int main(int argc, char * argv[])
{
	GReturn rc = GALIL_EXAMPLE_OK;
	char buf[G_SMALL_BUFFER];

	//var used to refer to a unique connection. A valid connection is nonzero.
	GCon g = 0;

	try
	{
		if (argc != 4) //Invalid number of arguments
		{
			cerr << "Incorrect number of arguments provided\n";
			cerr << "Usage: contour_example.exe <ADDRESS> <FileA> <FileB>\n";
			pause();
			return GALIL_EXAMPLE_ERROR;
		}

		char* fileA = argv[2];
		char* fileB = argv[3];
		char* address = argv[1];  //Retrieve address from command line
		e(GOpen(address, &g)); //Opens a connection at the provided address

		// Record user's training and plays back training through contour mode
		rc = contour(g, fileA, fileB);
	}
	catch (GReturn gr)
	{
		error(g, gr); //see examples.h for error handling
		pause();
		return GALIL_EXAMPLE_ERROR;
	}
    
    pause();
	return GALIL_EXAMPLE_OK;
}
/** @}*/

File: Auxiliary/gclib/examples/cs/examples/examples/contour_example.cs
/** @addtogroup cs_examples
  * @{
  */
  
/*! \file contour_example.cs
*
* See \link examples.Examples.Contour() Contour() \endlink for implementation of logic
* <br><br> For VB.NET, see definition in file contour_example.vb
*/
using System;
using System.Linq;

namespace examples
{
    /** @addtogroup cs_examples
    * @{
    */
	/// <summary>
	/// Record user's training and plays back training through contour mode.
	/// </summary>
	/// <remarks>
	/// The first argument should be the IP Address of a Galil controller.\n
	/// The second argument should be a path to a csv file holding positional data for the A axis.\n
	/// The third argument should be a path to a csv file holding positional data for the B axis.
	/// </remarks>
	/*! For VB.NET, see definition in file contour_example.vb */
    class Contour_Example
    {
        /// <summary>
        /// Main function for the contour example.
        /// </summary>
        /// <param name="args">An array of command line arguments.</param>
        /// <returns>The success status or error code of the function.</returns>
        /// <remarks>
        /// The first argument should be the IP Address of a Galil controller.\n
        /// The second argument should be a path to a text file where training for Axis A will be recorded.\n
        /// The third argument should be a path to a text file where training for Axis B will be recorded..
        /// </remarks>
        public static int Main(string[] args)
        {
            gclib gclib = new gclib();
            int rc = Examples.GALIL_EXAMPLE_OK;

            if (args.Count() != 3)
            {
                Console.WriteLine("Incorrect number of arguments provided");
                Console.WriteLine("Usage: record_position_example.exe <ADDRESS> <FILE A> <FILE B>");

                Console.Write("\nPress any key to close the example");
                Console.ReadKey();
                return Examples.GALIL_EXAMPLE_ERROR;
            }

            try
            {
                string address = args[0]; //Retrieve address from command line
                string fileA = args[1]; //Retrieve filepath from command line
                string fileB = args[2]; //Retrieve filepath from command line
                
                gclib.GOpen(address); //Opens connection at the provided address

                //Record user's training and play back training through contour mode
                rc = Examples.Contour(gclib, fileA, fileB);
            }
            catch(Exception ex)
            {
                Examples.PrintError(gclib, ex);
                rc = Examples.GALIL_EXAMPLE_ERROR;
            }
            finally
            {
                gclib.GClose();
            }

            Console.Write("\nPress any key to close the example");
            Console.ReadKey();

            return rc;
        }

    }
/** @}*/
}

/** @}*/